distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+
+# Defining options
+include(CMakeDependentOption)
+
+option(DISCOTEC_BUILD_MISSING_DEPS "Automatic fetch and build first layer of missing dependencies" ON)
+option(DISCOTEC_TIMING "Build with high-resolution timing" ON)
+if (DISCOTEC_TIMING)
+    target_compile_definitions(discotec PUBLIC TIMING)
+endif ()
+option(DISCOTEC_HDF5 "Interpolation output with HDF5" ON) # ON
+cmake_dependent_option(DISCOTEC_USE_HIGHFIVE "Use HighFive for HDF5" ON DISCOTEC_HDF5 OFF)
+option(DISCOTEC_UNIFORMDECOMPOSITION "Use uniform decomposition" ON) # TODO: @polinta: does not compile if off
+if (DISCOTEC_UNIFORMDECOMPOSITION)
+    target_compile_definitions(discotec PUBLIC UNIFORMDECOMPOSITION) # has to be PUBLIC for tests, rename to DISCOTEC_UNIFORMDECOMPOSITION?
+endif ()
+option(DISCOTEC_GENE "Build for GENE (as task library)" OFF) #TODO: handle
+if (DISCOTEC_GENE)
+    message(FATAL_ERROR "Building with GENE is not supported yet")
+endif ()
+option(DISCOTEC_OPENMP "Parallelize with OpenMP" OFF)
+if (DISCOTEC_OPENMP)
+    find_package(OpenMP REQUIRED)
+    target_link_libraries(discotec PUBLIC OpenMP::OpenMP_CXX)
+endif ()
+option(DISCOTEC_ENABLEFT "DisCoTec with algorithm-based fault tolerance" ON)
+if (DISCOTEC_ENABLEFT)
+    add_compile_definitions(ENABLEFT)
+endif ()
+option(DISCOTEC_DEBUG "Build with assertions and debug symbols" OFF) #TODO: Move to build type
+if (DISCOTEC_DEBUG OR CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+    target_compile_definitions(discotec PUBLIC DEBUG_OUTPUT)
+endif ()
+option(DISCOTEC_USE_LTO "Use link-time optimization" ON)
+if (DISCOTEC_USE_LTO)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT result OUTPUT output)
+    if (result)
+        set_property(TARGET discotec PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
+    else ()
+        message(FATAL "IPO is not supported: ${output}")
+    endif ()
+endif ()
+#TODO: find or fetch GENE if enabled
+
+option(DISCOTEC_OMITREADYSIGNAL "Omit ready signal" OFF)
+if (DISCOTEC_OMITREADYSIGNAL)
+    target_compile_definitions(discotec PRIVATE OMITREADYSIGNAL)
+endif ()
+
+option(DISCOTEC_USENONBLOCKINGMPICOLLECTIVE "Use non-blocking MPI collective operations" OFF)
+if (DISCOTEC_USENONBLOCKINGMPICOLLECTIVE)
+    target_compile_definitions(discotec PRIVATE USENONBLOCKINGMPICOLLECTIVE)
+endif ()
+
+#USE_NON_BLOCKING_COLLECTIVES #TODO: @polinta wird nirgends in SCONS definiert also ka wann definieren
+#ALT_LEVEL_VECTOR #TODO: @polinta wird nirgends in SCONS definiert also ka wann definieren
+#SIMFTMPI #TODO: @polinta: what is this? in confg.hpp the same is set no matter if defined or not ...
+#ISGENE #TODO: handle if access to GENE
+
+# Handle dependencies
+find_package(MPI REQUIRED)
+target_link_libraries(discotec PUBLIC MPI::MPI_CXX)
+
+include(FetchContent)
+
+find_package(Boost QUIET COMPONENTS serialization headers program_options unit_test_framework system ) #TODO: add all required components
```

Comment:
I've started to use boost-filesystem in some of the examples iirc

Reply:
added